src/UnigramWordPredictor.java

```diff
@@ -50,8 +51,18 @@ public UnigramWordPredictor(Tokenizer tokenizer) {
    */
   public void train(Scanner scanner) {
     List<String> trainingWords = tokenizer.tokenize(scanner);
+    neighborMap = new HashMap<>();
+    
+    for (int i = 0; i < trainingWords.size() - 1; i++) {
+      String word = trainingWords.get(i);
+      String nextWord = trainingWords.get(i + 1);
 
-    // TODO: Convert the trainingWords into neighborMap here
+      if (!neighborMap.containsKey(word)) {
+        neighborMap.put(word, new ArrayList<>());
+      }
```

Comment:
Nice! As an alternative, can you think about how you could write this using `putIfAbsent` or `computeIfAbsent` instead?